app/components-react/widgets/GameWidget.tsx

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import { IWidgetState, useWidget, WidgetModule } from './common/useWidget';
+import { WidgetLayout } from './common/WidgetLayout';
+import { $t } from '../../services/i18n';
+import { TextInput, ColorInput, createBinding, SliderInput } from '../shared/inputs';
+interface IGameWidgetState extends IWidgetState {
+  data: {
+    settings: {
+      decision_poll_timer: number;
+      trigger_command: string;
+      current_game: 'tic-tac-toe';
+      no_input_received_message: string;
+      restarting_game_message: string;
+      available_games: 'tic-tac-toe'[];
+      game_options: {
+        'tic-tac-toe': {
+          background_color: string;
+          border_color: string;
+          chat_marker_color: string;
+          ai_marker_color: string;
+          chat_win_marker_color: string;
+          ai_win_marker_color: string;
+          chat_won_game_message: string;
+          chat_lost_game_message: string;
+          draw_game_message: string;
+          chat_marker: string;
+          ai_marker: string;
+          game_ended_message_duration: number;
+          chat_turn_message: string;
+          ai_turn_message: string;
+          cannot_play_here: string;
+        };
+      };
+    };
+  };
+}
+
+function msToS(ms: number) {
+  return `${Math.floor(ms / 1000)}s`;
+}
+
+export function GameWidget() {
+  const { isLoading, bind } = useGameWidget();
+
+  return (
+    <WidgetLayout>
+      {!isLoading && (
+        <>
+          <SliderInput
+            label={$t('Chat Decision Time')}
+            {...bind.decision_poll_timer}
+            min={1000}
+            max={10000}
+            step={1000}
+            tipFormatter={msToS}
+          />
+          <TextInput label={$t('Trigger Command')} {...bind.trigger_command} />
+          <TextInput label={$t('No Input Recieved')} {...bind.no_input_received_message} />
+          <TextInput label={$t('Restarting Game')} {...bind.restarting_game_message} />
+          <GameOptions game="tic-tac-toe" />
+        </>
+      )}
+    </WidgetLayout>
+  );
+}
+
+export class GameWidgetModule extends WidgetModule<IGameWidgetState> {
+  bind = createBinding(
+    () => this.settings,
+    statePatch => this.updateSettings(statePatch),
+  );
+}
+
+function useGameWidget() {
+  return useWidget<GameWidgetModule>();
+}
+
+function GameOptions(p: { game: string }) {
+  const { settings, updateSettings } = useGameWidget();
+  const game = p.game;
+
+  function updateGameOption(key: string) {
+    return (value: any) => {
+      updateSettings({ game_options: { [game]: { [key]: value } } });
+    };
+  }
```

Comment:
You lose a type checking in this function. It accepts any string here

You may create a type `type TGameOptionName = keyof  %your_interface%`

Or you can use an `createBinding` util that generates types-safe `onChange` handlers for you

LMK if you need help here